packages/phone-number-privacy/signer/src/domain/domain.service.ts

```diff
@@ -7,58 +7,37 @@ import {
   domainQuotaStatusRequestSchema,
   DomainQuotaStatusResponse,
   DomainQuotaStatusResponseSuccess,
-  DomainResponse,
+  DomainRequest,
   domainRestrictedSignatureRequestSchema,
   DomainRestrictedSignatureResponse,
   DomainRestrictedSignatureResponseSuccess,
   DomainSchema,
   DomainState,
   ErrorMessage,
+  ErrorType,
+  KEY_VERSION_HEADER,
+  respondWithError,
   WarningMessage,
 } from '@celo/phone-number-privacy-common'
 import Logger from 'bunyan'
 import { Request, Response } from 'express'
 import { computeBlindedSignature } from '../bls/bls-cryptography-client'
 import { Counters } from '../common/metrics'
-import { getVersion } from '../config'
+import config, { getVersion } from '../config'
 import { getDatabase } from '../database/database'
-import { DOMAINS_STATES_COLUMNS, DomainStateRecord } from '../database/models/domainState'
+import { DomainStateRecord, DOMAINS_STATES_COLUMNS } from '../database/models/domainState'
 import {
   getDomainState,
   getDomainStateWithLock,
   insertDomainState,
   setDomainDisabled,
 } from '../database/wrappers/domainState'
 import { getKeyProvider } from '../key-management/key-provider'
+import { DefaultKeyName, Key } from '../key-management/key-provider-base'
 import { IDomainAuthService } from './auth/domainAuth.interface'
 import { IDomainService } from './domain.interface'
 import { IDomainQuotaService } from './quota/domainQuota.interface'
 
-// TODO: De-dupe with common package
-function respondWithError(
-  endpoint: DomainEndpoint,
-  res: Response<DomainResponse & { success: false }>,
-  status: number,
-  error: ErrorMessage | WarningMessage
-) {
-  const response: DomainResponse = {
-    success: false,
-    version: getVersion(),
-    error,
-  }
-
-  const logger: Logger = res.locals.logger
-
-  if (error in WarningMessage) {
-    logger.warn({ error, status, response }, 'Responding with warning')
-  } else {
-    logger.error({ error, status, response }, 'Responding with error')
-  }
-
-  Counters.responses.labels(endpoint, status.toString()).inc()
-  res.status(status).json(response)
-}
-
 export class DomainService implements IDomainService {
```

Comment:
I'm a bit confused about something here, but it's probably just my lack of understanding. I don't get how/if domains are scoped per application and how that's verified given that they can be disabled. Can't somebody just disable a domain used by another application at the moment?